### ISSUE TITLE
Support Snowflake directly for vector layers.

### ISSUE THREAD

**Issue by cmhh** [qgis/QGIS#52540]
### Feature description

The ability to drag-and-drop features directly from spatially-aware databases such as PostgreSQL / PostGIS or MSSQL is great, and it would be useful to be able to do the same thing with Snowflake tables which have a geometry or geography column.

### Additional context

Note that it is possible to use Snowflake via the existing ODBC functionality, and this is a plausible workaround in practice, albeit a little tougher to use for non-technical analysts.  However, in many enterprise settings, connecting to Snowflake using single sign-on is required, and the existing ODBC connectivity doesn't seem to support this.  That is, attempting to use an ODBC string or DSN where [`authenticator=externalbrowser`](https://docs.snowflake.com/en/user-guide/admin-security-fed-auth-use#setting-up-browser-based-sso) results in failure.  

Note also that I've raised this directly with reps from Snowflake, and they've told me they've raised a feature request, but that's all I know about it.

**Comment by Trippmadam** [comment]
Given that Snowflake offers two data types (geometry and geography) and support for any spatial reference system in conjunction with the geometry data type, it seems only logical for connection to GIS applications to be facilitated. Import of spatial data could be easier than it currently is as well.

**Comment by biellou** [comment]
I agree with @Trippmadam that supporting Snowflake would enable many business users to utilize QGIS. Currently, these users depend on commercial solutions for conducting sophisticated geospatial analysis.

**Comment by sfc-gh-obielov** [comment]
@cmhh @Trippmadam @biellou Feel free to try out the [QGIS-Snowflake Connector](https://github.com/snowflakedb/qgis-snowflake-plugin/tree/v0.7)

**Comment by cmhh** [comment]
> @cmhh @Trippmadam @biellou Feel free to try out the [QGIS-Snowflake Connector](https://github.com/snowflakedb/qgis-snowflake-plugin/tree/v0.7)

Thanks.  I tried to install this but it wasn't listed in the plugin repository.  I downloaded the zip and tried to install that, but it failed and then reported 'ModuleNotFoundError: No module named 'snowflake'.  It's possible that the snowflake connector failed to install because of my proxy, but I haven't looked into it yet.

**Comment by cmhh** [comment]
> @cmhh @Trippmadam @biellou Feel free to try out the [QGIS-Snowflake Connector](https://github.com/snowflakedb/qgis-snowflake-plugin/tree/v0.7)

I managed to install this plugin.  I had to manually install the `snowflake` and then restart QGIS.  But when I attempt to add a snowflake connection I don't end up with anything usable.  I have added a connection using SSO, but the connection form does not provide anywhere for me to enter a role or a schema.  I am told my connection is successful, but there's no tree of schemas or tables or anything I can select from.

**Comment by nyalldawson** [comment]
@cmhh can you report bugs in the plugin directly to the plugin repository (https://github.com/snowflakedb/qgis-snowflake-plugin) ? It's a third party plugin and support is given elsewhere.

**Comment by sfc-gh-obielov** [comment]
@cmhh I reported a new issue on your behalf: https://github.com/snowflakedb/qgis-snowflake-plugin/issues/13
